### PR TITLE
(WIP) glTF export: improve material export

### PR DESCRIPTION
**Pull request by querielo**
This pull request improves the material export in the glTF exporter.

* added support for exporting normal maps, metalness maps, and emissive maps.
* added support for exporting textures with texture transforms (tiling, offset, and rotation). 
* added support for exporting alpha test and alpha cutoff values for materials that use the "none" blend mode.

Please review. Thank you!

Screenshot 2023-11-10 at 14 18 22

### CODE REVIEW

File: extras/exporters/gltf-exporter.js [hunk 1]
```diff
@@ -350,13 +391,25 @@ class GltfExporter extends CoreExporter {
...
-                    material.alphaMode = "BLEND";
+                    material.alphaMode = 'BLEND';
+                } else if (blendType === BLEND_NONE) {
+                    if (alphaTest !== 0) {
+                        material.alphaMode = 'MASK';
+                        material.alphaCutoff = alphaTest;
+                    } else {
+                        // It is a default value, so we don't need to write it
+                        // material.alphaMode = 'OPAQUE';
+                    }
```
Comment (willeastcott) [inline extras/exporters/gltf-exporter.js:402]: Personally, I would just delete the else clause here.

**Review by willeastcott — APPROVED** [review]
This is a great contribution! It's great to see the glTF exporter get some love. Hey, @marklundin, got any idea how we can better make use of the glTF exporter in the Editor? Maybe a context menu option when right clicking on entities in the Hierarchy view?

**Comment by mvaligursky** [comment]
related to https://github.com/playcanvas/engine/issues/5001

File: extras/exporters/gltf-exporter.js (hunk 1 above)
Comment (querielo) [reply extras/exporters/gltf-exporter.js]: Done

**Comment by mvaligursky** [comment]
The code assumes the textures have their channels matching the glTF specification here. We should test for that (materials in PlayCanvas specify which channel is used) and only export the textures if the channels match.

Ideal solution in the future would be that if the channels do not match, we build the required textures from source textures.

Also, the normal map can have this format, which we should probably warn about and not export either, till we have code to convert it:

https://github.com/playcanvas/engine/blob/47d1f4c24b13ce15877b5cd79aaa204fa3abab92/src/platform/graphics/constants.js#L1049-L1056

The goal here being that we can only export valid glTF and print a warning otherwise.

**Comment by marklundin** [comment]
Yep great point @willeastcott, would love to surface this in a more intuitive way for users. It's a great piece of functionality and adds a lot of value. I'll have a think about this.

Great PR too @querielo 🔥

**Comment by querielo** [comment]
> Ideal solution in the future would be that if the channels do not match, we build the required textures from source textures.

I'll fix it later when I have free time.

**Comment by mvaligursky** [comment]
Please try to avoid make this PR larger and keep it focused on the material export improvements this PR is about, as this is getting hard to review now. Follow up work is awesome, but a separate PR please.

**Review by mvaligursky — APPROVED** [review]
Approving, tested and works well, well done and thanks!

**Pull request merged**